### PR TITLE
Sass

### PR DESCRIPTION
**Pull request by pmuir**

### CODE REVIEW

**Comment by pmuir** [comment]
I want reviews from Dan and perhaps Joshua.

Please see https://github.com/pmuir/ngx-widgets/blob/6a20d57e0fbc7b5ab6cefc8fca6837c88c2fd793/README.md#library-build which explains how to use this as a library.

I particularly want it checked that we don't have regressions in either the examples or in fabric8-planner.

File: README.md [hunk 1]
```diff
@@ -40,29 +40,38 @@ There are several services and a couple of models used by them available.
...
+
+To build  fabric8-planner as an npm library and embed it into a webapp such as
+fabric8-ui, you should:
+
+1. Run `npm run watch:library` in this directory. This will build fabric8-planner as
+a library and then set up a watch task to rebuild any ts, html and scss files you
+change.
+2. In the webapp into which you are embedding, run `npm link <path to fabric8-planner>/dist-watch`.
+This will create a symlink from `node_modules/fabric8-planner` to the `dist-watch` directory
+and install that symlinked node module into your webapp.
```
Comment (dlabrecq) [inline README.md]: Should we be linking planner or ngx-widgets here? I typically run 'npm link <path to ngx-widgets>/dist/'

File: README.md [hunk 2]
```diff
@@ -40,29 +40,38 @@ There are several services and a couple of models used by them available.
...
+3. Run your webapp in development mode, making sure you have a watch on `node_modules/fabric8-planner`
```
Comment (dlabrecq) [inline README.md]: How do we set the webapp to dev mode? I've seen a global ENV=development variable in webpack, so I suspect that's it. I'm also guessing the watch:library target is what we need to run in planner? 

Might be helpful to make it clear how to set dev mode and add a watch?

**Comment by pmuir** [comment]
Oops, that was a copy and paste error ;-)

File: README.md (hunk 1 above)
Comment (pmuir) [inline README.md]: ngx-widgets - copy and paste error, now fixed in readme.

File: README.md (hunk 2 above)
Comment (pmuir) [inline README.md]: in our projects, you just run npm start - webpack does all of that for you.

So in planner, run npm start if you want to run planner standalone, or npm run watch:library if you want to embed planner in to platform.

File: README.md [hunk 3]
```diff
@@ -40,29 +40,39 @@ There are several services and a couple of models used by them available.
...
+
+1. Run `npm run watch:library` in this directory. This will build ngx-widgets as
+a library and then set up a watch task to rebuild any ts, html and scss files you
+change.
+2. In the webapp into which you are embedding, run `npm link <path to ngx-widgets>/dist-watch`.
+This will create a symlink from `node_modules/fabric8-planner` to the `dist-watch` directory
+and install that symlinked node module into your webapp.
+3. Run your webapp in development mode, making sure you have a watch on `node_modules/ngx-widgets`
+enabled. To do this using a typical Angular Webpack setup, such as the one based on Angular Class,
+just run `npm start. You will have access to both JS sourcemaps and SASS sourcemaps if your webapp
```
Comment (joshuawilson) [inline README.md]: need a closing tic on "npm start"

File: config/webpack.common.js [hunk 4]
```diff
@@ -64,11 +70,59 @@ module.exports = {
           }
         ],
         exclude: [/\.spec\.ts$/]
+      },{
+        test: /\.css$/,
+        loader: extractCSS.extract({
```
Comment (joshuawilson) [inline config/webpack.common.js]: "loaders" should be "use"

File: config/webpack.common.js [hunk 5]
```diff
@@ -64,11 +70,59 @@ module.exports = {
...
+          fallback: "style-loader",
+          use: "css-loader?sourceMap&context=/"
+        })
+      }, {
+        test: /\.scss$/,
+        loaders: [
```
Comment (joshuawilson) [inline config/webpack.common.js]: "loaders" should be "use"

**Comment by joshuawilson** [comment]
@pmuir the webpack test config failed to compile but the tests pass. This gets us green but not a usable webpack.test.

File: config/webpack.demo.js [hunk 6]
```diff
@@ -56,19 +61,59 @@ module.exports = {
...
         test: /\.html$/,
         loader: 'html-loader'
-      },
-      {
+      }, {
         test: /\.css$/,
-        loaders: ['to-string-loader', 'css-loader']
-      },
-      {
+        loader: extractCSS.extract({
```
Comment (joshuawilson) [inline config/webpack.demo.js]: "loader/loaders" should be "use"
There are several of these.

File: config/webpack.test.js [hunk 7]
```diff
@@ -120,20 +123,66 @@ module.exports = function (options) {
...
+        /*
+         * to string and css loader support for *.css files
          * Returns file content as string
          *
-         * See: https://github.com/webpack/raw-loader
          */
         {
           test: /\.css$/,
-          use: ['to-string-loader', 'css-loader']
+          loaders: [
```
Comment (joshuawilson) [inline config/webpack.test.js]: "loader/loaders" should be "use"
There are several of these.

File: package.json [hunk 8]
```diff
@@ -149,6 +138,7 @@
     "karma-webpack": "2.0.2",
     "mocha": "3.2.0",
     "moment": "2.17.1",
+    "patternfly-sass": "^3.21.0",
```
Comment (joshuawilson) [inline package.json]: please don't leave the ^ in here, we will manually upgrade

**Comment by joshuawilson** [comment]
nevermind, the failing wepback was user error.

**Pull request merged**